File: apps/backend/src/main/java/hr/blitz/juice/domain/model/Reservation.java
package hr.blitz.juice.domain.model;

import hr.blitz.juice.audit.Auditable;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.time.LocalDate;
import java.util.List;

@Document(collection = "reservation")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Reservation extends Auditable<String> {

    @Id
    private String id;
    private User user;
    private Hall hall;
    private Instant date;
    private String timeFrom;
    private String timeTo;
    private List<User> players;
    private Terrain terrain;
    private boolean isPayed;
}

File: apps/frontend/src/models/entity/Reservation.ts
import Hall from "@/src/models/entity/Hall";
import User from "@/src/models/entity/User";
import Terrain from "@/src/models/entity/Terrain";

export default interface Reservation {
    id: string;
    user: User;
    hall: Hall;
    date: Date;
    timeFrom: string;
    timeTo: string;
    players: User[];
    terrain: Terrain;
    payed: boolean;
}